src/Pulsar.Client/Internal/DeadLetters.fs

```diff
@@ -0,0 +1,85 @@
+﻿namespace Pulsar.Client.Internal
+
+open System
+open Pulsar.Client.Api
+open Pulsar.Client.Common
+open System.Collections.Generic
+open FSharp.UMX
+open System.Threading.Tasks
+open Microsoft.Extensions.Logging
+open FSharp.Control.Tasks.V2.ContextInsensitive
+
+type internal DeadLettersProcessor
+    (policy: DeadLettersPolicy,
+     getTopicName: unit -> string,
+     getSubscriptionNameName: unit -> string,
+     createProducer: string -> Task<IProducer>) =
+
+    let store = Dictionary<MessageId, ResizeArray<Message>>()
+
+    let topicName =
+        if String.IsNullOrEmpty(policy.DeadLetterTopic) |> not then
+            policy.DeadLetterTopic
+        else
+            (sprintf "%s-%s-DLQ" (getTopicName()) (getSubscriptionNameName()))
+
+    let producer = lazy (
+
+        createProducer topicName
+    )
+
+    let sendMessage (builder : MessageBuilder) messageId  =
+        task {
+            let! p = producer.Value
+            let! _ = p.SendAsync(builder)
+            return ()
+        }
+
+    let toStoreFormat (messageId : MessageId) =
```

Comment:
Why is this method needed? Don't see such cast in Java code

Reply:
This method converts MessageId into key suitable for storing single and batch messages.
Notice. I does not copy java code line by line, because our project in other language. And original java code sometime looks unwanted tangled.